lib/ansible/modules/network/edgeos/edgeos_config.py

```diff
@@ -0,0 +1,284 @@
+                if item.startswith(entry):
+                    del commands[index]
+                    break
+            commands.append(item)
+        commands = ['set %s' % cmd.replace(' {', '') for cmd in commands]
+    else:
+        commands = str(candidate).split('\n')
+    return commands
+
+
+def get_candidate(module):
+    contents = module.params['src'] or module.params['lines']
+
+    if module.params['lines']:
+        contents = '\n'.join(contents)
+
+    return config_to_commands(contents)
+
+
+def diff_config(commands, config):
+    config = [str(c).replace("'", '') for c in config.splitlines()]
+
+    updates = list()
+    visited = set()
+
+    for line in commands:
+        item = str(line).replace("'", '')
+
+        if not item.startswith('set') and not item.startswith('delete'):
+            raise ValueError('line must start with either `set` or `delete`')
+
+        elif item.startswith('set') and item not in config:
+            updates.append(line)
+
+        elif item.startswith('delete'):
+            if not config:
+                updates.append(line)
+            else:
+                item = re.sub(r'delete', 'set', item)
+                for entry in config:
+                    if entry.startswith(item) and line not in visited:
+                        updates.append(line)
+                        visited.add(line)
+
+    return list(updates)
+
+
+def sanitize_config(config, result):
+    result['filtered'] = list()
+    for regex in CONFIG_FILTERS:
+        for index, line in enumerate(list(config)):
+            if regex.search(line):
+                result['filtered'].append(line)
+                del config[index]
+
+
+def run(module, result):
+    # get the current active config from the node or passed in via
+    # the config param
+    config = module.params['config'] or get_config(module)
+
+    # create the candidate config object from the arguments
+    candidate = get_candidate(module)
+
+    # create loadable config that includes only the configuration updates
+    commands = diff_config(candidate, config)
+    sanitize_config(commands, result)
+    result['commands'] = commands
+
+    commit = not module.check_mode
+    comment = module.params['comment']
+
+    if commands:
+        load_config(module, commands, commit=commit, comment=comment)
+        if result.get('filtered'):
+            result['warnings'].append('Some configuration commands were '
+                                      'removed, please see the filtered key')
+        result['changed'] = True
+
+
+def main():
+    argument_spec = dict(
+        src=dict(type='path'),
+        lines=dict(type='list'),
+        match=dict(default='line', choices=['line', 'none']),
+        comment=dict(default=DEFAULT_COMMENT),
+        config=dict(),
+        backup=dict(default=False, type='bool'),
+        save=dict(default=False, type='bool'),
+    )
+
+    argument_spec.update(vyos_argument_spec)
```

Comment:
Actually, if we lose this line (and the import of `vyos_argument_spec`, and the module doc fragment, and the action plugin, and so on), we shouldn't actually need to change the module_utils, though it might be useful for clarity, or in case more problematic changes happen